crates/uv-distribution/src/metadata/build_requires.rs

```diff
@@ -203,3 +204,93 @@ impl BuildRequires {
+                    .as_ref()
+                    .and_then(|tool| tool.uv.as_ref())
+                    .and_then(|uv| uv.index.as_deref())
+                    .unwrap_or(&[]);
+
+                let empty_sources = BTreeMap::default();
+                let project_sources = workspace
+                    .pyproject_toml()
+                    .tool
+                    .as_ref()
+                    .and_then(|tool| tool.uv.as_ref())
+                    .and_then(|uv| uv.sources.as_ref())
+                    .map(ToolUvSources::inner)
+                    .unwrap_or(&empty_sources);
+
+                // Lower each package's extra build dependencies
+                let mut result = ExtraBuildDependencies::default();
+                for (package_name, requirements) in extra_build_dependencies {
+                    let lowered: Vec<uv_pep508::Requirement<VerbatimParsedUrl>> = requirements
+                        .into_iter()
+                        .flat_map(|requirement| {
+                            let requirement_name = requirement.name.clone();
+                            let extra = requirement.marker.top_level_extra_name();
+                            let group = None;
+                            LoweredRequirement::from_requirement(
+                                requirement,
+                                None,
+                                workspace.install_path(),
+                                project_sources,
+                                project_indexes,
+                                extra.as_deref(),
+                                group,
+                                index_locations,
+                                workspace,
+                                None,
+                            )
+                            .map(
+                                move |requirement| match requirement {
+                                    Ok(requirement) => Ok(requirement.into_inner().into()),
+                                    Err(err) => Err(MetadataError::LoweringError(
+                                        requirement_name.clone(),
+                                        Box::new(err),
+                                    )),
+                                },
+                            )
+                        })
+                        .collect::<Result<Vec<_>, _>>()?;
+                    result.insert(package_name, lowered);
+                }
+                Ok(Self {
+                    extra_build_dependencies: result,
+                })
+            }
+            SourceStrategy::Disabled => {
+                // Without source resolution, just return the dependencies as-is
+                Ok(Self {
+                    extra_build_dependencies,
+                })
+            }
+        }
+    }
+
+    /// Create from pre-lowered dependencies (for non-workspace contexts).
+    pub fn from_lowered(extra_build_dependencies: ExtraBuildDependencies) -> Self {
```

Comment:
Should this just be a `From<ExtraBuildDependencies>` impl?

Reply:
Yeah I think a dedicated method makes sense here.